pkg/controller/data_download_controller.go

```diff
@@ -485,10 +485,6 @@ func (r *DataDownloadReconciler) SetupWithManager(mgr ctrl.Manager) error {
 						return false
 					}
 
-					if newObj.Status.Phase != v1.PodRunning {
```

Comment:
because we want to watch all kinds of status pods instead of only keep running pods, then use `findSnapshotRestoreForPod` to check unrecoverable pods, and make them fail early.